src/context/TodoContext.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  createContext,
+  useContext,
+  useState,
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+} from 'react';
+
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+interface TodoContextType {
+  todos: Todo[];
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+  filter: Filter;
+  setFilter: Dispatch<SetStateAction<Filter>>;
+  error: string | null;
+  setError: Dispatch<SetStateAction<string | null>>;
+}
+
+const TodoContext = createContext<TodoContextType>({
+  todos: [],
+  setTodos: () => {},
+  filter: Filter.ALL,
+  setFilter: () => {},
+  error: null,
+  setError: () => {},
+});
+
+export const useTodoContext = () => useContext(TodoContext);
+
+export const TodoContextProvider: React.FC<PropsWithChildren> = ({
+  children,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
```

Comment:
The 'useState' for 'todos' should be initialized with the initial state coming from the API.